src/main/com/fulcrologic/fulcro/components.cljc

```diff
@@ -799,13 +805,16 @@
   "Given a state map and a query, returns a state map with the query normalized into the database. Query fragments
   that already appear in the state will not be added.  Part of dynamic query implementation."
   [state-map query]
-  (let [new-state (normalize-query-elements state-map query)
-        new-state (if (nil? (::queries new-state))
-                    (assoc new-state ::queries {})
-                    new-state)
-        top-query (link-query query)]
-    (if-let [queryid (some-> query meta :queryid)]
-      (util/deep-merge {::queries {queryid {:query top-query :id queryid}}} new-state)
+  (let [queryid       (some-> query meta :queryid)
+        component-key (class->registry-key (some-> query meta :component))
+        query'        (vary-meta query dissoc :queryid :component)
```

Comment:
Why is this necessary? (clearing the metadata)...oh, is this so it does not end up in state map?

Reply:
That was my thought, keeping metadata out of state. If you think its OK to leave it then I will update...